statsmodels/base/model.py

```diff
@@ -1311,6 +1313,129 @@ def f_test(self, r_matrix, cov_p=None, scale=1.0, invcov=None):
                              invcov=invcov, use_f=True)
         return res
 
+    @staticmethod
+    def _is_lr_test_invalid(model, attribute, expected_value):
+        """
+        Likelihood test is not valid when:
+            1) cov_type of model is set to nonrobust
+            2) .fit(method=REML) in case of MixedLM
+
+        Parameters
+        ----------
+        model: Result instance
+
+        attribute: string
+            Attribute to be looked up in the input model.
+
+        expected_value: string
+            Expected value of the input attribute.
+
+        Returns
+        -------
+        True if the test is invalid
+        """
+        if getattr(model, attribute, expected_value) != expected_value:
+            return True
+        return False
+
+    def compare_lr_test(self, restricted, large_sample=False):
+        """
+        Likelihood ratio test to test whether restricted model is correct
+
+        Parameters
+        ----------
+        restricted : Result instance
+            The restricted model is assumed to be nested in the current model.
+            The result instance of the restricted model is required to have two
+            attributes, residual sum of squares, `ssr`, residual degrees of
+            freedom, `df_resid`.
+
+        large_sample : bool
+            Flag indicating whether to use a heteroskedasticity robust version
+            of the LR test, which is a modified LM test.
+
+        Returns
+        -------
+        lr_stat : float
+            likelihood ratio, chisquare distributed with df_diff degrees of
+            freedom
+        p_value : float
+            p-value of the test statistic
+        df_diff : int
+            degrees of freedom of the restriction, i.e. difference in df between
+            models
+
+        Notes
+        -----
+
+        The exact likelihood ratio is valid for homoskedastic data, and is
+        defined as
+
+        .. math:: D=-2\\log\\left(\\frac{\\mathcal{L}_{null}}
+           {\\mathcal{L}_{alternative}}\\right)
+
+        where :math:`\mathcal{L}` is the likelihood of the model. With :math:`D`
+        distributed as chisquare with df equal to difference in number of
+        parameters or equivalently difference in residual degrees of freedom.
+
+        The large sample version of the likelihood ratio is defined as
+
+        .. math:: D=n s^{\\prime}S^{-1}s
+
+        where :math:`s=n^{-1}\\sum_{i=1}^{n} s_{i}`
+
+        .. math:: s_{i} = x_{i,alternative} \\epsilon_{i,null}
+
+        is the average score of the model evaluated using the residuals from
```

Comment:
Where do the equations and this part come from?
My guess is that they are not correct in this case, or it doesn't describe the LR test that is actually used.
It looks like it's from the score/Lagrange multiplier test which is not used in the method below.